crypto/s390xcap.c

```diff
@@ -64,4 +113,475 @@ void OPENSSL_cpuid_setup(void)
     sigaction(SIGFPE, &oact, NULL);
     sigaction(SIGILL, &oact, NULL);
     sigprocmask(SIG_SETMASK, &oset, NULL);
+
+    OPENSSL_s390x_functions();
```

Comment:
Any specific reason for splitting OPENSSL_s390x_facilities? Since facilities zero whole thing, one can say that it's only appropriate that it initializes whole thing...

Reply:
Ah! Right :-)

Reply:
It probably makes sense to move the zeroing of the query function's part over to OPENSSL_s390x_functions though.